x/tss/keeper/keeper.go

```diff
@@ -225,6 +225,55 @@ func (k Keeper) GetMaxSimultaneousSignShares(ctx sdk.Context) int64 {
 	return shares
 }
 
+// GetSignedBlocksWindow returns the max simultaneous number of sign shares
+func (k Keeper) GetSignedBlocksWindow(ctx sdk.Context) int64 {
+	var window int64
+	k.params.Get(ctx, types.KeySignedBlocksWindow, &window)
+
+	return window
+}
+
+// ValidatorMissedTooManyBlocks returns true if the given validator address missed too many blocks within
+// the block window specifiec by this module
+func (k Keeper) ValidatorMissedTooManyBlocks(ctx sdk.Context, address sdk.ConsAddress) bool {
+	signedBlocksWindow := k.GetSignedBlocksWindow(ctx)
+	missedBlocks := k.getMissedBlocksCount(ctx, address)
+	maxMissedBlocks := k.GetMaxMissedBlocksPerWindow(ctx)
+
+	missedBlocksThreshold := utils.Threshold{Numerator: missedBlocks, Denominator: signedBlocksWindow}
+	return missedBlocksThreshold.GTE(maxMissedBlocks)
+}
+
+// returns the number of blocks signed w.r.t. this module's signed blocks window parameter
+func (k Keeper) getMissedBlocksCount(ctx sdk.Context, address sdk.ConsAddress) int64 {
+	counter := int64(0)
+	tssWindow := k.GetSignedBlocksWindow(ctx)
+	slasherWindow := k.slasher.SignedBlocksWindow(ctx)
+	signInfo, ok := k.slasher.GetValidatorSigningInfo(ctx, address)
+
+	if !ok {
+		return 0
+	}
+
+	indexOffset := int64(0)
+	window := slasherWindow
+	if slasherWindow > tssWindow {
```

Comment:
What happens if `slasherWindow < tssWindow`?

Reply:
We need to look within the boundaries of the shorter window. Otherwise we would be trying to fetch information from outside the signed block window. Unless I am missing something?

Reply:
@haiyizxx @cgorenflo Any thought on this guys? I understand that the information is lost but find it weird about this implication.

Reply:
I guess we could add additional checks when validating the genesis file?

Reply:
I like that idea. Looks like we can access params from other modules during genesis validation, nice.

Reply:
Another possibility is to extend the documentation for the behaviour of this function

Reply:
Easy to miss the documentation imo

Reply:
After discussing this with Haiyi and Sammy, we decided that its better to just re-implement the whole mechanism on the tss module instead of re-purposing cosmos code. This will allow us to not worry about the cosmos own window. But since this demands an extensive refactoring, we decided to pause this PR for now. We will get back to this in another release.

Reply:
we will return to this in the future: https://github.com/axelarnetwork/axelar-core/issues/1207